### PR TITLE
[G7T5-178] Update main README with feature descriptions

### PR DESCRIPTION
**Pull request by jonathantan1425**
# Description

Fixes [Jira ticket #G7T5-178](https://is212g7t5.atlassian.net/browse/G7T5-178)

https://github.com/is212g7t5/spm/tree/jon/G7T5-178#readme

List all acceptance criterias of user story and indicate which has been solved previously and which is solved currently
- [x] Overall README contains sufficient descriptions and illustrations for users to use the product independently. (solved)
- [x] No application logic change 

## Type of change

Please delete options that are not relevant.

- [x] Refactor

# How Has This Been Tested?
- [x] I have added the corresponding testcases into the [testcase document](https://docs.google.com/spreadsheets/d/1QOyUN_kFN0fddLkjAQz2DK3jou3cWdN9CJsNbl3v0Kg/edit#gid=0)

Please indicate the testcase ID you have added or are using

- [x] NA, no application logic change

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+
+#### Deactivate
+
+Selecting the `Deactivate` option on a job role brings up a confirmation window to "soft-delete" a job role (i.e. set the job role to inactive). This would prevent Staff users from viewing the respective job role.
+
+<img src="assets/hr/deactivate_job.png" alt="HR Deactivate Job" />
+
+### Manager Features
+
+Currently, none of the first release features are unique to Managers, and therefore no notable features are available under this role. Future features such as viewing the skills of their team members will be added in subsequent releases.
```
Comment (deadline2359) [inline README.md:276]: Currently, none of the first release features **is** unique to Managers, and therefore no notable features **is** available under this role. Future features such as viewing the skills of their team members will be added in subsequent releases.

File: README.md [hunk 2]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+
+In the create learning journey page, the job details are displayed at the top, followed by the respective skills, similar to the update page earlier.
+
+<img src="assets/staff/create_learning_journey.png" alt="Staff Create Learning Journey" />
+
+Selecting on a skill required brings up a list of courses, which can then be added to the learning journey.
+
+<img src="assets/staff/create_learning_journey_skill_course.png" alt="Staff Create Learning Journey Skill Course" />
+
+Once at least 1 course is added to the learning course, the `Create Learning Journey` button becomes active and the user can create the learning journey with the associated courses.
```
Comment (deadline2359) [inline README.md:197]: added to the learning **_journey_**

File: README.md [hunk 3]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+
+It is also on this page where new learning journeys are created. The `+ New Learning Journey` button in the navbar will reroute the user to this page so that they can specify the job role that they want to create a new learning journey for. This is then completed by clicking on the `Create Learning Journey` button in the desired job role.
+
+##### Creating a Learning Journey
+
+In the create learning journey page, the job details are displayed at the top, followed by the respective skills, similar to the update page earlier.
+
+<img src="assets/staff/create_learning_journey.png" alt="Staff Create Learning Journey" />
+
+Selecting on a skill required brings up a list of courses, which can then be added to the learning journey.
```
Comment (deadline2359) [inline README.md:193]: Selecting on a **_required_** skill

File: README.md [hunk 4]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+
+<img src="assets/staff/courses.png" alt="Staff Courses" />
+
+#### Job Roles
+
+Navigate to the job roles page (https://spm-g7t5.netlify.app/jobs) or click on the `Job Roles` tab in the navbar to view all active job roles. Clicking on any of the job roles reveals the job role's descriptions and its accompanying skills.
+
+<img src="assets/staff/jobs.png" alt="Staff Jobs" />
+
+It is also on this page where new learning journeys are created. The `+ New Learning Journey` button in the navbar will reroute the user to this page so that they can specify the job role that they want to create a new learning journey for. This is then completed by clicking on the `Create Learning Journey` button in the desired job role.
```
Comment (deadline2359) [inline README.md:185]: **_for_** the desired job role

File: README.md [hunk 5]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+Selecting one of the four possible states simulates the access control permissions of the particular state. Each of these states are also pegged to a specific user (specified by `id`) and the default can be changed in `/app/src/contexts/UserContext.jsx`, as defined in the `USER_TYPES` const.
+
+Alternatively, the states can be changed through the session storage of the web browser followed by a web browser refresh. For example, the `userId` and `user` can be changed as seen below.
+<img src="assets/session_storage.png" alt="Session Storage" />
+
+By extension, this means that users with multiple roles, should log in as each of their role in order to access that role's features.
+
+> For e.g., a HR user should log in as a `Staff` to view their learning journey planning, but log in as a `HR` to edit the jobs available.
+
+This design choice was made to facilitate a cleaner user interface (not having both "user" and "admin" related permissions in the same location), which will ultimately be supported by a role-separated login page in subsequent releases.
```
Comment (deadline2359) [inline README.md:129]: **_clean_** user interface

File: README.md [hunk 6]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+<img src="assets/access_control_toggle.png" alt="Access Control Toggle" />
+
+Selecting one of the four possible states simulates the access control permissions of the particular state. Each of these states are also pegged to a specific user (specified by `id`) and the default can be changed in `/app/src/contexts/UserContext.jsx`, as defined in the `USER_TYPES` const.
+
+Alternatively, the states can be changed through the session storage of the web browser followed by a web browser refresh. For example, the `userId` and `user` can be changed as seen below.
+<img src="assets/session_storage.png" alt="Session Storage" />
+
+By extension, this means that users with multiple roles, should log in as each of their role in order to access that role's features.
+
+> For e.g., a HR user should log in as a `Staff` to view their learning journey planning, but log in as a `HR` to edit the jobs available.
```
Comment (deadline2359) [inline README.md:127]: as `HR` (no a)

File: README.md [hunk 7]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
...
+
+Access control is currently being segmented into 4 possible states (`Not logged in`, `Staff`, `HR` amd `Manager`). Use the dropdown menu in the top-right corner of the navbar to toggle between the 4 states.
+<img src="assets/access_control_toggle.png" alt="Access Control Toggle" />
+
+Selecting one of the four possible states simulates the access control permissions of the particular state. Each of these states are also pegged to a specific user (specified by `id`) and the default can be changed in `/app/src/contexts/UserContext.jsx`, as defined in the `USER_TYPES` const.
+
+Alternatively, the states can be changed through the session storage of the web browser followed by a web browser refresh. For example, the `userId` and `user` can be changed as seen below.
+<img src="assets/session_storage.png" alt="Session Storage" />
+
+By extension, this means that users with multiple roles, should log in as each of their role in order to access that role's features.
```
Comment (deadline2359) [inline README.md:125]: this means that users, with multiple roles (add a comma)

File: README.md [hunk 8]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
 <!-- USAGE EXAMPLES -->
 ## Usage
 
-To be updated nearer to first release.
+### Access Control
+
+Access control is currently being segmented into 4 possible states (`Not logged in`, `Staff`, `HR` amd `Manager`). Use the dropdown menu in the top-right corner of the navbar to toggle between the 4 states.
+<img src="assets/access_control_toggle.png" alt="Access Control Toggle" />
+
+Selecting one of the four possible states simulates the access control permissions of the particular state. Each of these states are also pegged to a specific user (specified by `id`) and the default can be changed in `/app/src/contexts/UserContext.jsx`, as defined in the `USER_TYPES` const.
```
Comment (deadline2359) [inline README.md:120]: Selecting one of the **_three_**

 Each of these states **_is_**

File: README.md [hunk 9]
```diff
@@ -102,10 +104,169 @@ Refer to the relevant CI pipeline scripts in `/.github/workflows`.
 <!-- USAGE EXAMPLES -->
 ## Usage
 
-To be updated nearer to first release.
+### Access Control
+
+Access control is currently being segmented into 4 possible states (`Not logged in`, `Staff`, `HR` amd `Manager`). Use the dropdown menu in the top-right corner of the navbar to toggle between the 4 states.
```
Comment (deadline2359) [inline README.md:117]: Toggle between **_3_** states

**Review by deadline2359 — APPROVED** [review]

**Review by quinncheong — APPROVED** [review]

**Pull request merged**